x-pack/plugins/observability_ai_assistant/server/functions/esql/correct_common_esql_mistakes.ts

```diff
@@ -92,6 +115,19 @@ export function correctCommonEsqlMistakes(content: string, log: Logger) {
           case 'STATS':
             formattedCommand = replaceAsKeywordWithAssignments(formattedCommand);
             break;
+
+          case 'SORT':
+            formattedCommand = replaceFunctionsinSortWithStats(formattedCommand);
+            break;
+
+          case 'KEEP':
+            if (array.length > index + 1) {
+              if (array[index + 1].match(/^([A-Za-z]+)/)?.[1] === 'SORT') {
+                formattedCommand = verifySortColumnInKeep(formattedCommand, array[index + 1]);
+                break;
+              }
```

Comment:
Maybe a better approach here would be to do this for the `SORT` command, and then iterate back over the commands until we find a `KEEP` command? I can write that code and push it if you want to.

Reply:
sure! I did it in this order because I was not sure how to modify previous commands